### PR TITLE
optimize: fix build action warning

### LINKED ISSUE #5943: github actions is deprecated and will be disabled soon
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description

image
https://github.com/seata/seata/actions/runs/6574549668

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

Minimal yet complete reproducer code (or URL to code):

### Ⅵ. Environment:

- JDK version(e.g. `java -version`):
- Seata client/server version:  
- Database version:
- OS(e.g. `uname -a`):
- Others:

### PR DESCRIPTION
**Pull request by leizhiyuan**
- [x] I have registered the PR [changes](../changes).

### Ⅱ. Does this pull request fix one issue?

fixes #5943

### Ⅳ. Describe how to verify it

run the action

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
       # step 2
       - name: "Set up QEMU"
         id: qemu
-        uses: docker/setup-qemu-action@v1
+        uses: docker/setup-qemu-action@v3
```
Comment (slievrly) [inline .github/workflows/build.yml:57]: The build_arm64-binary action only works on push events, so there is no way to confirm that this issue has been resolved until this PR merge. Therefore, I propose to temporarily add build_arm64-binary action to pull request in this PR and remove it after confirming that there is no problem.

Comment (leizhiyuan) [reply .github/workflows/build.yml:57]: https://github.com/leizhiyuan/seata/actions/runs/6582808710/job/17884865374 
I will test in my fork

Comment (leizhiyuan) [reply .github/workflows/build.yml:57]: https://github.com/leizhiyuan/seata/actions/runs/6582999974 (to speed up)

the waring has been resolved, To accelerate the verification process, I mock some command scripts that are unrelated to warnings.

Comment (slievrly) [reply .github/workflows/build.yml:57]: ok, it seems to be working fine.

**Review by slievrly — APPROVED** [review]
LGTM

**Review by wangliang181230 — APPROVED** [review]
LGTM

**Comment by wangliang181230** [comment]
Does the `develop` branch also have this problem?

**Pull request merged**